sympy/solvers/solvers.py

```diff
@@ -2740,9 +2740,20 @@ def equal(expr1, expr2):
                 return list(ordered(set(sol)))
 
         elif lhs.is_Mul and rhs.is_positive:
-            llhs = expand_log(log(lhs))
+            even_degrees = [i for i in degree_list(lhs) if i%2 == 0]
```

Comment:
Why the check is in `solve`. Shouldn't it be in `bivariate.py` somewhere in `solve_lambert`?
The reason being the fact that we will again require it for `solveset`.

Reply:
Actually solve itself was creating illusion of preprocessing the equation by `expand_log(log(lhs))`. But now it is in bivariate. Please check latest code I will be attatching a proof of correctness also soon.